components/suggestion/command_provider/app_provider.tsx

```diff
@@ -75,16 +77,17 @@ export default class AppCommandProvider extends Provider {
                 switch (suggestion.type) {
                 case COMMAND_SUGGESTION_USER:
                     element.push(AtMentionSuggestion);
-                    return suggestion.item! as UserProfile;
+                    return {...suggestion.item! as UserProfile, type: COMMAND_SUGGESTION_USER};
                 case COMMAND_SUGGESTION_CHANNEL:
                     element.push(ChannelMentionSuggestion);
-                    return {channel: suggestion.item! as Channel};
+                    return {channel: suggestion.item! as Channel, type: COMMAND_SUGGESTION_CHANNEL};
                 default:
                     element.push(CommandSuggestion);
                     return {
                         ...suggestion,
                         Complete: '/' + suggestion.Complete,
                         Suggestion: '/' + suggestion.Suggestion,
+                        type: Constants.Integrations.COMMAND,
```

Comment:
The value of `suggestion` here isn't necessarily a command. It could be a select option or boolean for example.  I think we should go with no type here, since `suggestion.default` will take care of this. What do you think @sinansonmez?

To handle the case of subcommands, I'm we can put the `type` here: https://github.com/mattermost/mattermost-webapp/blob/cca778f2e6dbc929ba0fb90c15cde9ef20b6c4ca/components/suggestion/command_provider/app_command_parser/app_command_parser.ts#L1584-L1590

Supporting other kinds of the fields in `app_command_parser.ts` is probably out of scope of this PR.

Reply:
This suggestion object specifically on line 90 is the one place I was suggesting we remove the type. All of the other `command` places that were previously in the PR can be reintroduced. Also the change to line 1589 of `app_provider` is still relevant. Does this make sense @sinansonmez?

Reply:
It makes sense but we will get the same results (i.e. the message will be `SUGGESTION` which is the default message) when I keep the types for command but change it to `suggestion.default`. Anyhow it makes total sense and easier to follow for future devs if we add types in the source (i.e. `app_command_parser` and `command_provider`  @sinansonmez